### ISSUE TITLE
An include from a subdirectory doesn't prevent an orphan warning

### ISSUE THREAD

**Issue by uellue** [sphinx-doc/sphinx#6813]
**Describe the bug**

Including a RST file from a subdirectory using `.. include::` doesn't disable the orphan warning `WARNING: document isn't included in any toctree`.

**To Reproduce**

This occurred on Windows 10, Unix behavior is untested.

* Build a minimum Sphinx setup, for example with `sphinx-quickstart`
* Create a subfolder in the source dir
* Create a RST file in that subfolder
* Include that file in `index.rst` using `.. include:: subdir/file.rst`
* Build the project

**Expected behavior**

There's no warning about an orphaned file since the file is included. This works if the file is in the same folder.

**Your project**

[sphinx-issue.zip](https://github.com/sphinx-doc/sphinx/files/3836383/sphinx-issue.zip)

**Screenshots**

Not applicable

**Environment info**
- OS: Win 10
- Python version: Python 3.7.4
- Sphinx version: sphinx-build 2.2.1
- Sphinx extensions:  None
- Extra tools: None

**Additional context**

Not applicable

**Comment by uellue** [comment]
As a workaround, the folder can be excluded using the `exclude_patterns` setting.

**Comment by tk0miya** [comment]
On my local, macOS and Linux, it seems working fine. So I guess this bug came from directory separator. Could you build it again with a backslash for including path?

**Comment by uellue** [comment]
@tk0miya Changing the path separator to a backslash produces the same result on my system, i.e. an orphan warning. Still, probably the path separator playing a role somewhere if it works on Unix. :-)

**Comment by tk0miya** [comment]
Thank you for confirm. I'll try to find the reason later.

**Comment by tk0miya** [comment]
My friend, @drillan reproduces the warning and runs debugger (thanks!).
It seems `env.included` contains a string based on windows path separator (backslash)!
As a result, `autogenerated/test.rst` is treated as orphaned document.
```
> c:\users\driller\work\sphinx-issue\venv\lib\site-packages\sphinx\environment\__init__.py(638)check_consistency()
-> logger.warning(__('document isn\'t included in any toctree'),
(Pdb) p docname
'autogenerated/test'
(Pdb) p self.included
defaultdict(<class 'set'>, {'index': {'samedir', 'autogenerated\\test'}})
```

**Comment by tk0miya** [comment]
I just posted #6836 to fix this. On @drillan's local, it works fine. So I'll merge it after a while.

**Comment by uellue** [comment]
Great, thank you for taking care of this! :-)

**Comment by tk0miya** [comment]
Fixed by #6836.
Thank you for reporting!
